### PR TITLE
Updates `Account` component to act based on the connection status

### LINKED ISSUE #89: Tell user to switch network while on wrong network
### Problem
When the user is on the wrong network, the `LoginModal` shows a wallet to connect, even if the wallet is already connected.

### Solution
The modal should display a message asking the user to switch to the proper network.

### LINKED ISSUE #133: Account component should behave depending on the connection status
**Problem**
When the provider is set and the user clicks on the Account component, we still show the set provider options.

**Solution**
We should show something relevant in these situations:
- If the user is properly logged in, the modal doesn't add any value
- If the user is on the wrong network, we should ask to switch network

**Implementation idea**
Our Web3Provider could have `ConnectionStatus` prop on the state that would tell the components how to act depending on the situation

**Screenshots**
Screen Shot 2020-08-17 at 17 17 40
Screen Shot 2020-08-17 at 17 17 56

### PR DESCRIPTION
**Pull request by farinalvaro**
This PR closes #133 and closes #89

### CODE REVIEW

File: src/components/organisms/Account.tsx [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+  setProvider: (provider: EProvider) => Promise<void>
+  web3?: Web3
+  networkInfo?: NetworkInfo
+  account?: string
+  availableProviders?: ProviderInfo[]
+  requiredNetworkId?: number
+  onCorrectNetworkMessage?: string
+  onNetworkMismatchMessage?: string
+  noNetworkMessage?: string
+}
```
Comment (jurajpiar) [inline src/components/organisms/Account.tsx:25]: that's rather a lot of props. can it not be reduced somehow?

Comment (jurajpiar) [reply src/components/organisms/Account.tsx:25]: Actually, looking at how it is used, can the <Web3Provider.Consumer> be created inside the Account component instead?

**Review by jurajpiar — APPROVED** [review]

File: src/components/organisms/Account.tsx (hunk 1 above)
Comment (farinalvaro) [reply src/components/organisms/Account.tsx:25]: Good point, I will try to combine some of the props :)
About using the Context inside the component, I'm trying not to do that so users can use the library independently of the context

Comment (farinalvaro) [reply src/components/organisms/Account.tsx:25]: Resolving as we agreed to postpone this refactor :)

**Pull request merged**